apps/andi/test/integration/andi_web/live/edit_live_view_test.exs

```diff
@@ -638,4 +638,93 @@ defmodule AndiWeb.EditLiveViewTest do
       refute Enum.empty?(find_elements(html, ".data-dictionary-remove-field-editor__error-msg--visible"))
     end
   end
+
+  describe "finalize form" do
+    setup do
+      dataset =
+        TDG.create_dataset(%{
+          technical: %{
+            cadence: "1 1 1 * * *"
+          }
+        })
+
+      {:ok, andi_dataset} = Datasets.update(dataset)
+      [dataset: andi_dataset]
+    end
+
+    data_test "quick schedule #{schedule}", %{conn: conn, dataset: dataset} do
+      assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
+
+      render_click([view, "finalize_form_editor"], "quick_schedule", %{"schedule" => schedule})
+      html = render(view)
+
+      assert expected_crontab == get_crontab_from_html(html)
+
+      where([
+        [:schedule, :expected_crontab],
+        ["hourly", "0 0 * * * *"],
+        ["daily", "0 0 0 * * *"],
+        ["weekly", "0 0 0 * * 0"],
+        ["monthly", "0 0 0 1 * *"],
+        ["yearly", "0 0 0 1 1 *"]
+      ])
+    end
+
+    test "set schedule manually", %{conn: conn, dataset: dataset} do
+      assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
+
+      render_click([view, "finalize_form_editor"], "set_schedule")
+      html = render(view)
+
+      assert dataset.technical.cadence == get_crontab_from_html(html)
+      assert Enum.empty?(find_elements(html, ".finalize-form__error-msg--visible"))
+    end
+
+    test "handles five-character cronstrings", %{conn: conn} do
+      dataset = TDG.create_dataset(%{technical: %{cadence: "4 2 7 * *"}})
+      {:ok, _} = Datasets.update(dataset)
```

Comment:
Interesting I never thought of asserting like this. Have you seen the test fail if it returns {:error, _}.

Reply:
Yeah, it's not an assertion but there will be an error if it fails